src/games/strategy/triplea/ui/menubar/ExportMenu.java

```diff
@@ -168,27 +173,48 @@ public String getDescription() {
       // A small warning so users will not over-write a file,
       if (f.exists()) {
         final int choice =
-            JOptionPane.showConfirmDialog(null, "A file by that name already exists. Do you wish to over write it?",
+            JOptionPane.showConfirmDialog(frame, "A file by that name already exists. Do you wish to over write it?",
                 "Over-write?", JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE);
         if (choice != JOptionPane.OK_OPTION) {
           return;
         }
       }
+
       final File file = f;
-      final Runnable t = () -> {
-        if (saveScreenshot(node, file, frame, gameData)) {
-          JOptionPane.showMessageDialog(null, "Map Snapshot Saved", "Map Snapshot Saved",
-              JOptionPane.INFORMATION_MESSAGE);
+      final SwingWorker<?, ?> task = new SwingWorker<Void, Void>() {
+        @Override
+        protected Void doInBackground() throws Exception {
+          saveScreenshot(node, file, frame, gameData);
+          return null;
+        }
+
+        @Override
+        protected void done() {
+          try {
+            get();
+            SwingUtilities.invokeLater(() -> {
+              JOptionPane.showMessageDialog(frame, "Map Screenshot Saved", "Map Screenshot Saved",
+                  JOptionPane.INFORMATION_MESSAGE);
+            });
+          } catch (final ExecutionException e) {
+            SwingUtilities.invokeLater(() -> {
+              JOptionPane.showMessageDialog(frame, e.getCause().getMessage(), "Error Saving Screenshot",
+                  JOptionPane.OK_OPTION);
+            });
+          } catch (final InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
Interesting read, sounds like it advocates rethrowing the interrupt.

My reading/understanding of how to handle is to abort/stop any on-going operations, clean-up and return. The JVM has received a terminate signal, so things 'should' be shutting down.  

Rather than be consistent in handling, I'd prefer to discuss what the better handling is, and try to create a library function that we can apply consistently. The library function will then make this a no-brainer of an operation, which is good for going over and writing lots of code : )

- does seem like interrupting the current thread manually is not needed. I recall reading it's not the most predictable thing. 
- we could rethrow the interrupted exception. Wrapping it probably is called for so that we don't have to have checked exceptions in the entire stack. Alternatively we do a 'return' from method.